src/contexts/EmployeeContext.tsx

```diff
@@ -7,69 +7,156 @@ import {
+  const updateLoading = (loader: string, value: boolean) => {
+    setLoading((prev) => {
+      return { ...prev, [loader]: value };
+    });
   };
 
+  const [searchParams, setSearchParams] = useSearchParams({"page":"1","sortBy":"id","sortDir":"asc"});
+  const updateSearchParams = (params: {
+    page?: string;
+    sortBy?: string;
+    sortDir?: string;
+  }) => {
+    setSearchParams(
+      {
+        ...Object.fromEntries(searchParams.entries()),
+        ...params,
+      },
+      { replace: true }
+    );
+  };
+  const [filters, setFilters] = useState(initialContextValues.filters);
   const updateFilters = (newFilters: {
-    skills?: string[];
+    skills?: ISkill[];
     search?: string[];
   }) => {
     if (newFilters.skills || newFilters.search) {
       setFilters((prev) => ({ ...prev, ...newFilters }));
     }
   };
 
+  const [employeesData, setEmployeesData] = useState(
+    initialContextValues.employeesData
+  );
   const updateEmployeesData = (newData: IEmployee[]) => {
     setEmployeesData(newData);
   };
 
-  const updateIdToDelete = (id: number) => {
-    setIdToDelete(id);
+  const [skills, setSkills] = useState(initialContextValues.skills);
+  const { employeeId } = useParams();
+  const fetchEmployeesData = async () => {
+    try {
+      updateLoading("isTableLoading", true);
+      const searchSortBy = searchParams.get("sortBy") || "id";
+      const searchSortDir = searchParams.get("sortDir") || "asc";
+      const searchPage = searchParams.get("page") || "1";
+      const response = await getData(`/employee`, {
+        params: {
+          limit: rowsPerPage,
+          offset: (Number(searchPage) - 1) * rowsPerPage,
+          sortBy: searchSortBy ? searchSortBy : "id",
+          sortDir: searchSortDir ? searchSortDir : "asc",
+        },
+      });
+      let employeesData = response.data.data.employees.map(
+        (employeeData: any) => {
+          return {
+            ...employeeData,
+            lastName: employeeData.lastName ? employeeData.lastName : "",
+            skills: employeeData.skills ? employeeData.skills : [],
+            role: employeeData.role?.role || "N/A",
+            department: employeeData.department?.department || "N/A",
+          };
+        }
+      );
+      updateEmployeesData(employeesData);
+      count = response.data.data.count;
+    } catch (error) {
+      toast.error(`Employees details could not be fetched from server.`, {
+        position: "top-center",
+        autoClose: 5000,
+        hideProgressBar: false,
+        closeOnClick: true,
+        pauseOnHover: true,
+        draggable: true,
+        progress: undefined,
+        theme: "colored",
+      });
+    } finally {
+      updateLoading("isTableLoading", false);
+    }
   };
+  useEffect(() => {
+    if (!employeeId) {
+      fetchEmployeesData();
+    }
+  }, [searchParams]);
+  useEffect(() => {
+    const fetchSkills = async () => {
```

Comment:
Why don't you write api calls can be in a different file? If i wanted to call the api in another component code duplication will happen right?

Reply:
Made changes to fetch function.